File: multi-requests.py
#!/bin/python3
import sys
from requests_futures.sessions import FuturesSession

PISTON_URL = 'http://127.0.0.1:2000/api/v2/execute'
FILE_NAME = sys.argv[1]
[_, LANGUAGE] = sys.argv[1].split('.')
SOURCE_CODE_FILE = open(FILE_NAME)
SOURCE_CODE = SOURCE_CODE_FILE.read()
SOURCE_CODE_FILE.close()
NO_REQUESTS = int(sys.argv[2]) if len(sys.argv) >= 3 else 16
VERSION = sys.argv[3] if len(sys.argv) >= 4 else '*'
RUNTIME = sys.argv[4] if len(sys.argv) >= 5 else None

session = FuturesSession()
futures = [session.post(PISTON_URL, json={
    'language': LANGUAGE,
    'version': VERSION,
    'runtime': RUNTIME,
    'files': [{ 'content': SOURCE_CODE }]
}) for i in range(NO_REQUESTS)]

File: multi-requests.js
const fs = require('fs');

(async () => {
    const file_name = process.argv[2];
    const [_, language] = file_name.split('.');
    const URL = 'http://127.0.0.1:2000/api/v2/execute';
    const file = fs.readFileSync(file_name);
    const requests = [];
    const no_requests = parseInt(process.argv[3]);

    for (let i = 0; i < no_requests; ++i) {
        requests.push(
            fetch(URL, {
                headers: {
                    'CONTENT-TYPE': 'application/json'
                },
                method: 'POST',
                body: JSON.stringify({
                    language,
                    version: '*',
                    files: [{ content: file.toString() }]
                })
            }).then((res) => res.json())
        );
    }

    const responses = await Promise.all(requests);
    console.log(responses);
})();
